### ISSUE TITLE
Email sent to your gmail about issue

### ISSUE THREAD

**Issue by tradingplatfroms** [Hephyrius/binance_futures_bot#3]
Kindly check gmail urgent .

**Comment by Nai-mul** [comment]
really need him... i am also in trouble
Please help me. 

 python3 bot.py
Bot Starting
Encountered Exception ('ExecuteError', '[Executing] -2015: Invalid API-key, IP, or permissions for action, request ip: 103.126.149.19')
Encountered Exception ('ExecuteError', '[Executing] -2015: Invalid API-key, IP, or permissions for action, request ip: 103.126.149.19')

**Comment by prafulrana** [comment]
Running into same issue

**Comment by Nai-mul** [comment]
> Running into same issue

if you found any solution please don't forget to help me :(

**Comment by prafulrana** [comment]
Yes. Figured it out. My Futures account was not enabled. Go to binanance.com/en/futures to enable it.

**Comment by Nai-mul** [comment]
> Yes. Figured it out. My Futures account was not enabled. Go to binanance.com/en/futures to enable it.

hey bro, looks like its very interesting bot. but the problem is bot closing position very by trilling stop triggered .
and please give me your telegram

**Comment by Chris6622** [comment]
Hi all,

Having the same Error message as mentioned above
I have checked my binance account settings! looks all good. Futurestrading is enabled and API key and secret is new created and pasted into file keys.json

Have you any idea to solve the problem

Encountered Exception ('ExecuteError', '[Executing] -2015: Invalid API-key, IP, or permissions for action, request ip: 103.126.149.19')

**Comment by Hephyrius** [comment]
Hey All, apologies for not seeing these messages until now. Check the security in your API settings. You may need to add your IP address to the allowed devices.

**Comment by mhgutier** [comment]
> > Yes. Figured it out. My Futures account was not enabled. Go to binanance.com/en/futures to enable it.
> 
> hey bro, looks like its very interesting bot. but the problem is bot closing position very by trilling stop triggered .
> and please give me your telegram... my telegram @Newsrell

hi, 
handle_signal seems like making side = -1 regardless buy or sell

add this in hND


> > Yes. Figured it out. My Futures account was not enabled. Go to binanance.com/en/futures to enable it.
> 
> hey bro, looks like its very interesting bot. but the problem is bot closing position very by trilling stop triggered .
> and please give me your telegram... my telegram @Newsrell

hi,
edit the handle signal
    
def handle_signal(client, std, market="ETHUSDT", leverage=3, order_side="BUY", stop_side="SELL", _callbackRate=2.0):
    initialise_futures(client, _market=market, _leverage=leverage)
    qty = calculate_position(client, market, _leverage=leverage)
    
    enablePrint(std)
    execute_order(client, _qty=qty, _side=order_side, _market=market)
    blockPrint()
    
    in_position = True
    
    if order_side == "BUY" :
        side = 1
    elif order_side == "SELL" :
        side = -1
    
    market_price = get_market_price(client, _market=market)
    
    singlePrint(f"{order_side}: {qty} ${market_price} using x{leverage} leverage", std)

**Comment by Nai-mul** [comment]
after adding that i am getting error msg 
if order_side == "BUY" :
    side = 1
elif order_side == "SELL" :
    side = -1


like that

Encountered Exception ('ExecuteError', "[Executing] -4061: Order's position side does not match user's setting.")

**Comment by Hephyrius** [comment]
> after adding that i am getting error msg
> if order_side == "BUY" :
> side = 1
> elif order_side == "SELL" :
> side = -1
> 
> like that
> 
> Encountered Exception ('ExecuteError', "[Executing] -4061: Order's position side does not match user's setting.")

are you using some sort of hedge mode?

**Comment by Nai-mul** [comment]
i turned off hedge mode when I start it. Then two things are happening together. 
![IMG_20210522_194944](https://user-images.githubusercontent.com/64520358/119228914-213a5b00-bb37-11eb-8101-fece2b91f8bb.jpg)
![IMG_20210522_194920](https://user-images.githubusercontent.com/64520358/119228919-239cb500-bb37-11eb-8981-5c27fd1726e7.jpg)

**Comment by Nai-mul** [comment]
Mt bot setting is

{
	"market": "1000SHIBUSDT",
	"leverage": "3",
	"trading_periods": "5m,15m",
	"margin_type": "CROSSED",
	"trailing_percentage": "2.0"
}

And wallet balance 3usdt

**Comment by shin31120** [comment]
I found what is problem.
https://help.quantower.com/connections/connection-to-binance-futures/errors-with-binance-connection#error-invalid-api-key-ip-or-permissions-for-action
Here is link!.
